Remove SwiftOnoneSupport & Comments

ollieatkinson opened the pull request:
Fix #5
Fix #6

Todo:
- Write tests for `GenerateCommand.swift`
- Ensure we are happy with regex replace for `RemoveComments.swift`. Currently this works fine, but deletes new lines between code, potentially consider new regex or figure out how to strip leading whitespace when using `sourcekitten`.

ollieatkinson committed:
Remove SwiftOnoneSupport & Comments

Fix #5
Fix #6

paulsamuels commented on Sources/GuiseFramework/APIGenerator.swift:
```diff
@@ -1,10 +1,24 @@
 import Foundation
 import SourceKittenFramework
 
+protocol SourceKittenRequest {
```
Is this change related? It looks like this is the start of making `APIGenerator` testable but need further work

ollieatkinson replied:
No, this needs moving into it's own request - good spot!

paulsamuels commented on Sources/GuiseFramework/TextProcessors/RemoveComments.swift:
```diff
@@ -0,0 +1,13 @@
+  func process(input: String) throws -> String {
+
+    return input.split(separator: "\n").filter({ line in
+      line.range(of: " *\\/\\/\\/", options: .regularExpression) == nil
```
- Could tidy the regex up a tiny bit with " *\\/{3}"
- It looks like the `split` that is dropping the new lines - can add `omittingEmptySubsequences: false`

ollieatkinson committed:
fixup! Remove SwiftOnoneSupport & Comments

ollieatkinson committed:
Simplify regex and ensure new lines are retained

Why:
New lines were being removed when splitting the text, we can avoid
this by omittingEmptySubsequences.

This change simpifies the regex and adds the parameter
omittingEmptySubsequences to the split function.

paulsamuels commented on Tests/GuiseFrameworkTests/TextProcessors/RemoveCommentsTests.swift:
```diff
@@ -0,0 +1,28 @@
+    let input = """
+/// This is a doc comment for my structure
+struct MyStructure {
+  /// This is a doc comment for my property
```
Is it worth adding a new line into this test case so we can catch if new lines are accidentally stripped?

ollieatkinson replied:
Yep! Sounds like a good idea

ollieatkinson committed:
fixup! Remove SwiftOnoneSupport & Comments

paulsamuels commented on Tests/GuiseFrameworkTests/TextProcessors/RemoveSwiftOnoneSupportTests.swift:
```diff
@@ -0,0 +1,22 @@
+    
+    let systemUnderTest = RemoveSwiftOnoneSupport()
+    
+    XCTAssertEqual(try systemUnderTest.process(input: "\nimport SwiftOnoneSupport\nimport Foundation"), "\nimport Foundation", "SwiftOnoneSupport should be removed, but it was not.")
```
This test case is not realistic as the file might not start with a new line. I guess that also raises that this could potentially not delete the import if it was the first line for some reason. Should it be `import SwiftOnoneSupport\n` instead?

paulsamuels reviewed (approved):
One remaining question but looks good 🚗

Sherlouk commented on Sources/GuiseFramework/CLI/GenerateCommand.swift:
```diff
@@ -77,6 +81,21 @@ struct GenerateCommand: CommandProtocol {
       }
     }
     
+    
```
nit: double whitespace

Sherlouk commented on Sources/GuiseFramework/CLI/GenerateCommand.swift:
```diff
@@ -77,6 +81,21 @@ struct GenerateCommand: CommandProtocol {
+    } catch {
+      return .failure(.unexpectedError(error: error))
+    }
+    
```
General comment, not a blocker for this but if this function continues to expand with additional functionality then we should probably think about breaking it down a little. You've already got comments breaking it into logical chunks!

ollieatkinson committed:
fixup! Remove SwiftOnoneSupport & Comments
